client/src/partials/budget/edit/edit_budget.html

```diff
@@ -3,160 +3,150 @@
+                <span class="glyphicon glyphicon-plus-sign"></span> {{ "BUDGET.EDIT.CREATE" | translate }}
+              </button>
+            </h3>
+          </div>
+        </div>
+      </div> <!-- end create budget row -->
+      <div class="row panel panel-primary" ng-if="session.found" style="margin: 4px 4px auto 4px">
+        <div class="panel-heading">
+          <b>{{ "BUDGET.EDIT.BUDGET_DETAILS" | translate }}</b>
+        </div>
+        <div class="col-xs-5">
+          <form name="budget" style="margin-top: 10px">
+            <table class="table form-bhima-border" summary="{{ 'EDIT.BUDGET.EDIT' | translate }}">
+              <thead>
+              <tr>
+                <th colspan="5">
+                  <h4>{{ "BUDGET.EDIT.FISCAL_YEAR_LABEL" | translate }} {{ session.fiscal_year.fiscal_year_txt }}</h4>
+                </th>
+              </tr>
+              <tr>
+                <th>{{ "BUDGET.EDIT.PERIOD" | translate }}</th>
+                <th>{{ "BUDGET.EDIT.START"  | translate }}</th>
+                <th>{{ "BUDGET.EDIT.END"    | translate }}</th>
+                <th>{{ "BUDGET.EDIT.BUDGET" | translate }}</th>
+                <th>{{ "BUDGET.EDIT.FREEZE" | translate }}</th>
+              </tr>
+              </thead>
+              <tbody>
+                <tr ng-repeat="budget in budgets.data">
+                  <td>{{ budget.period_number }}</td>
+                  <td>{{ budget.period_start | date }}</td>
+                  <td>{{ budget.period_stop | date }}</td>
+                  <td><input type="number" 
+                       ng-model="budget.budget" 
+                       ng-change="recompute()" 
+                       ng-focus="startEditing(budget)"
+                       ng-blur="endEditing(budget)"
+                       min="0.0" 
+                       ng-disabled="session.totalBudget <= 0.0 || (session.autoAdjust && (budget.freeze && !budget.editing))"
+                       >
+                  </td>
+                  <td style="text-align: center">
+                    <input type="checkbox" 
+                     ng-model="budget.freeze">
+                  </td>
+                </tr>
+                <tr>
+                  <td colspan="2" style="vertical-align: middle">
+                    <b>{{ "BUDGET.EDIT.AUTO_ADJUST" | translate }}&nbsp;&nbsp;
+                    <input type="checkbox" 
+                     style="margin: 0"
+                     ng-disabled="session.totalBudget <= 0.0"
+                     ng-change="recompute()"
+                     ng-model="session.autoAdjust"></b>
+                  </td>
+                  <td style="vertical-align: middle">
+                    <span class="pull-right">{{ "BUDGET.EDIT.TOTAL" | translate }}</span>
+                  </td>
+                  <td style="vertical-align: middle">
+                    <input type="number"
+                     ng-class="{'form-bhima-red-border' : session.totalBudget <= 0}"
```

Comment:
There are other ways to detect and handle errors in forms using a controller and a `formController.input.$error`-type properties.  We should research this in the future so that we don't re-implement the same logic in each controller.